src/mame/drivers/seattle.cpp

```diff
@@ -519,6 +520,10 @@ void seattle_state::machine_reset()
 
 	if (m_board_config == SEATTLE_WIDGET_CONFIG)
 		widget_reset();
+
+	// Set the disk dma transfer speed
+	auto* hdd = subdevice<ide_hdd_device>(PCI_ID_IDE":ide:0:hdd");
+	hdd->set_dma_transfer_time(attotime::from_usec(15));
```

Comment:
Should this be a slot option configuration fragment?

Reply:
Not sure what you mean, do you have an example?
The hdd doesn't get resolved until after the machine initialization so it can't go into config.

Reply:
Devices can’t be added after machine initialisation.  You can add configuration fragments for slot options with `device_slot_interface::set_option_machine_config`.

Reply:
This doesn't work, I am missing something.
void seattle_state::cfg_dma_15us(device_t* device)
{
	dynamic_cast<device_slot_interface&>(*device->subdevice(PCI_ID_IDE":ide:0:hdd")).set_dma_transfer_time(attotime::from_usec(15));
}

Reply:
Have you looked at how other drivers do it, e.g. for configuring 512 bytes per sector for CD-ROM drives?

Reply:
Another try, closer but doesn't work
	subdevice<bus_master_ide_controller_device>(PCI_ID_IDE":ide")->slot(0).set_option_machine_config("hdd", hdd_config);
}

void seattle_state::hdd_config(device_t* device)
{
	dynamic_cast<ide_hdd_device*>(device)->set_dma_transfer_time(attotime::from_usec(15));
}

Reply:
nvm got something working